jetty-io/src/main/java/org/eclipse/jetty/io/SelectorManager.java

```diff
@@ -212,11 +213,10 @@ private ManagedSelector chooseSelector(SelectableChannel channel)
         // The ++ increment here is not atomic, but it does not matter,
         // so long as the value changes sometimes, then connections will
         // be distributed over the available selectors.
-        long s = _selectorIndex++;
-        int index = (int)(s % getSelectorCount());
+        int index = _selectorIndex.updateAndGet(i->(i+1)%_selectors.length);
```

Comment:
Not liking the lambda here, it is capturing and hence allocating an object every time. `incrementAndGet()` in one line and `%` on another line, but then we need the AtomicLong above.

Reply:
Making new connections should not be too frequent, and already creates a lot of objects... but let's avoid more if we can.